src/test/java/racingcar/CarTest.java

```diff
@@ -0,0 +1,47 @@
+package racingcar;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import racingcar.car.Car;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class CarTest {
+
+    @DisplayName("자동차를 생성하면 위치가 0인 자동차가 생성된다.")
+    @Test
+    void createCarTest() {
+        // given // when
+        Car car = Car.createCar();
+
+        // then
+        Assertions.assertThat(car.getCurrentLocation()).isEqualTo(0);
+    }
+
+    @DisplayName("4 이상의 숫자가 주어지면 자동차의 위치가 1 증가한다.")
+    @Test
+    void carMoveTestWithEnoughPower() {
+        // given
+        Car car = Car.createCar();
+
+        // when
+        car.move(TestEngine.createCarEngine(4));
+
+        // then
+        assertThat(car.getCurrentLocation()).isEqualTo(1);
+    }
+
+    @DisplayName("4 미만의 숫자가 주어지면 자동차의 위치는 변화하지 않는다.")
+    @Test
+    void carMoveTestWithNotEnoughPower() {
+        // given
+        Car car = Car.createCar();
+
+        // when
+        car.move(TestEngine.createCarEngine(3));
```

Comment:
이 방법 외에도 간단하게 익명클래스를 만들어 활용하시는 방법도 고민해보셨으면 좋을것 같아요 :)

```suggestion
        car.move(()-> true);
```

현재는 전진하는 조건이 랜덤 엔진에서 `DEFAULT_MOVE_ENERGY` 상수로 관리하고 있는데, 이 조건을 바꾸면 모킹하고 있는 `TestEngine.createCarEngine(3)` 이 부분도 일일히 모두 변경해야하는 구조로 보여져요!

이 fake 객체는 움직이는 조건을 판단하는데에는 관심이 없고 `전진한다. 하지않는다.` 만 관심이 있으니 이렇게 표현해보시는것도 좋을것 같다는 의견입니다 🙂

Reply:
말씀하신대로 적용해보았습니다. 
항상 테스트를 작성할 때 세부 조건에 의존해서 테스트가 깨지는 경우가 많은데 그런 고민 계속 하겠습니다.